packaging/deb/package.targets

```diff
@@ -0,0 +1,198 @@
+    <MakeDir Directories="$(debLayoutSamples)" />
+    <MakeDir Directories="$(debLayoutDocs)" />
+    
+    <!-- Copy files to debian layout -->
+    <ItemGroup>
+        <HFFiles Include="$(InputRoot)/**/*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(HFFiles)" DestinationFolder="$(debLayoutPackageRoot)"/>
+
+    <!-- Replace config json variables -->
+    <WriteLinesToFile
+      File="$(debLayoutDirectory)$(ConfigJsonName)"
+      Lines="$([System.IO.File]::ReadAllText('$(ConfigJsonFile)').Replace('%25HOSTFXR_BRAND_NAME%25', '$(HostFxrBrandName)').Replace('%25SHARED_HOST_DEBIAN_VERSION%25', '$(HostVersion)').Replace('%25HOSTFXR_NUGET_VERSION%25', '$(HostVersion)').Replace('%25HOSTFXR_DEBIAN_PACKAGE_NAME%25', '$(DebPackageName)')
+      )"
+      Overwrite="true" />
+
+    <Exec Command="deb-tool -i $(debLayoutDirectory) -o $(debIntermediatesDir) -n $(DebPackageName) -v $(DebPackageVersion)"
+          WorkingDirectory="$(consumingProjectDirectory)" />
+
+  </Target>
+
+  <Target Name="GenerateSharedFrameworkDeb">
+    <PropertyGroup>
+      <DebPackageName>$(SharedFxDebPkgName)</DebPackageName>
+      <DebPackageVersion>$(HostVersion)</DebPackageVersion>
+      <InputRoot>$(SharedHostPublishRoot)</InputRoot>
+      <DebFile>$(SharedHostInstallerFile)</DebFile>
+      <ConfigJsonName>dotnet-sharedframework-debian_config.json</ConfigJsonName>
+      <ConfigJsonFile>$(debPackaginfConfigPath)$(ConfigJsonName)</ConfigJsonFile>
+      <debIntermediatesDir>$(PackagesIntermediateDir)$(DebPackageName)/$(DebPackageVersion)</debIntermediatesDir>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <debLayoutDirectory>$(debIntermediatesDir)/debianLayoutDirectory/</debLayoutDirectory>
+      <debLayoutAbsolute>$(debLayoutDirectory)$</debLayoutAbsolute>
+      <debLayoutPackageRoot>$(debLayoutDirectory)package_root</debLayoutPackageRoot>
+      <debLayoutSamples>$(debLayoutDirectory)samples</debLayoutSamples>
+      <debLayoutDocs>$(debLayoutDirectory)docs</debLayoutDocs>
+    </PropertyGroup>
+
+    <RemoveDir Condition="Exists('$(debIntermediatesDir)')" Directories="$(debIntermediatesDir)" />
+    <MakeDir Directories="$(debIntermediatesDir)" />
+
+    <!-- Create empty debian layout -->
+    <RemoveDir Condition="Exists('$(debLayoutDirectory)')" Directories="$(debLayoutDirectory)" />
+    <MakeDir Directories="$(debLayoutDirectory)" />
+    <MakeDir Directories="$(debLayoutAbsolute)" />
+    <MakeDir Directories="$(debLayoutPackageRoot)" />
+    <MakeDir Directories="$(debLayoutSamples)" />
+    <MakeDir Directories="$(debLayoutDocs)" />
+    
+    <!-- Copy files to debian layout -->
+    <ItemGroup>
+        <SHFiles Include="$(InputRoot)/**/*" />
+        <SHManpages Include="$(ManPagesDir)/**/*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(SHFiles)" DestinationFolder="$(debLayoutPackageRoot)"/>
+    <Copy SourceFiles="@(SHManpages)" DestinationFolder="$(debLayoutDocs)"/>
+
+    <!-- Replace config json variables -->
+    <WriteLinesToFile
```

Comment:
looks way better than using string,Replace. Have you thought if moving it to buildtools?